bundled/boost-1.62.0/include/boost/signals2/detail/auto_buffer.hpp

```diff
@@ -304,6 +304,7 @@ namespace detail
             (*this).~auto_buffer();
             buffer_   = new_buffer;
             members_.capacity_ = new_capacity;
+            size_ = std::max(static_cast<decltype(size_)>(0),size_);
```

Comment:
Can you explain why this change is correct? I mean, I have no idea of what the code does, it just looks funny to add a line in a random place in boost :-)